examples/kyb-app/src/components/organisms/DynamicUI/Page/Page.tsx

```diff
@@ -0,0 +1,66 @@
+import { UIElement, UIPage } from '@app/domains/collection-flow';
+import { AnyChildren, AnyObject } from '@ballerine/ui';
+import { useStateManagerContext } from '@app/components/organisms/DynamicUI/StateManager/components/StateProvider';
+import { useMemo } from 'react';
+import { useRuleExecutor } from '@app/components/organisms/DynamicUI/hooks/useRuleExecutor';
+import { useDynamicUIContext } from '@app/components/organisms/DynamicUI/hooks/useDynamicUIContext';
+import { PageContext } from '@app/components/organisms/DynamicUI/Page/types';
+import { ErrorField } from '@app/components/organisms/DynamicUI/rule-engines';
+import { pageContext } from './page.context';
+import { usePageErrors } from '@app/components/organisms/DynamicUI/Page/hooks/usePageErrors';
+import { usePageResolverContext } from '@app/components/organisms/DynamicUI/PageResolver/hooks/usePageResolverContext';
+
+const { Provider } = pageContext;
+
+export interface PageProps {
+  page: UIPage;
+  children: AnyChildren;
+}
+
+export const Page = ({ page, children }: PageProps) => {
+  const { pages } = usePageResolverContext();
+  const definition = useMemo(() => {
+    const definition: UIElement<AnyObject> = {
+      type: 'page',
+      name: page.name,
+      options: {},
+    };
+
+    return definition;
+  }, [page]);
+
+  const rules = useMemo(() => (page.pageValidator ? [page.pageValidator] : []), [page]);
```

Comment:
Using `useMemo` for non-computational derived state may introduce more overhead than being an optimization.

Reply:
`useMemo` in this case is used as memoization of array to prevent additional computations in hooks which rely on this value. Otherwise it will cause recomputations on every re-render if value will be provided like this `useSomeHook([val])`

Hook which using it accepts rules as an array so instead of generating additional hook just for this case useMemo is used.

`useMemo` is mandatory here.